onnxruntime/core/providers/cuda/reduction/reduction_ops.cc

```diff
@@ -604,9 +606,15 @@ Status ReduceKernel<allow_multi_axes>::ComputeImpl(OpKernelContext* ctx, cudnnRe
                                        prepare_reduce_metadata));
 
   Tensor* Y = ctx->Output(0, prepare_reduce_metadata.squeezed_output_dims);
+  bool fast_reduction = fast_reduction_;
```

Comment:
can this be be done in the ReduceSum Kernel construction time? 
fast_reduction_ should be populated according to the value of GetUseDeterministicCompute() 

This way, we only need to set the fast_reduction once.

Reply:
kernel constructor is called at program loading time. use_deterministic_compute is set when a training session is created. I may add GetUseDeterministicCompute() to OpKernelContext to avoid a cast at each run.

Reply:
coded a bit, maybe it is not good idea to touch OpKernelContext because it is exposed via API.